snakemake/io.py

```diff
@@ -1327,10 +1327,29 @@ class Resources(Namedlist):
     pass
 
 
-class Log(Namedlist):
+class LogStreamError(AttributeError):
     pass
 
 
+class Log(Namedlist):
+    def streams(self):
+        _streams = {
+            name: str(value)
+            for name, value in self.items()
+            if name in ("std", "stderr", "stdout")
+        }
+        if "std" in _streams and ("stderr" in _streams or "stdout" in _streams):
+            raise LogStreamError("Can't combine std stream with stdout/stderr.")
+        if any(path for path in _streams.values() if " " in path):
+            raise LogStreamError("Std output stream has more than one entry.")
+
+        for path in _streams.values():
+            dirname = os.path.dirname(path)
+            if any(dirname) and not os.path.exists(dirname):
+                os.makedirs(dirname)
+        return {name: open(value, "w") for name, value in _streams.items()}
```

Comment:
The log files should be opened in append mode, such that multiple shell commands below each other don't overwrite their output.

Reply:
Sure thing